### ISSUE TITLE
false negative for disallowed-methods

### ISSUE THREAD

**Issue by xxchan** [rust-lang/rust-clippy#9558]
### Summary

```rust
use futures::stream::{empty, select_all};

fn main() {
    let _ = select_all(vec![empty::<()>()]);
}
```
with `clippy.toml`
```toml
disallowed-methods = [{ path = "futures::stream::select_all" }]
```
produces no warnings.

Since `futures::stream` is re-export of `futures_util::stream`, I also tried that but clippy didn't work neither.

### Lint Name

disallowed-methods

### Version

```text
rustc 1.64.0 (a55dd71d5 2022-09-19)
binary: rustc
commit-hash: a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52
commit-date: 2022-09-19
host: aarch64-apple-darwin
release: 1.64.0
LLVM version: 14.0.6
```
